### PR TITLE
Add more unit tests for azure services

### PR DESCRIPTION
**Pull request by alexander-demicev**
This PR adds more unit tests for azure services. There are no logical changes in cluster create/update actions.

### CODE REVIEW

File: pkg/aks/check.go [hunk 1]
```diff
@@ -113,7 +113,7 @@ var regionToOmsRegionMap = map[string]string{
 	"chinanorth2": "chinaeast2",
 }
 
-func CheckLogAnalyticsWorkspaceForMonitoring(ctx context.Context, client services.WorkplacesClientInterface,
+func checkLogAnalyticsWorkspaceForMonitoring(ctx context.Context, client services.WorkplacesClientInterface,
```
Comment (mjura) [inline pkg/aks/check.go:116]: `CheckLogAnalyticsWorkspaceForMonitoring` is public function, I was thinking that we want keep them starting with UpperCase

**Review by mjura — APPROVED** [review]
Looks ok, only small nitpick

File: pkg/aks/check.go (hunk 1 above)
Comment (alexander-demicev) [reply pkg/aks/check.go:116]: I noticed it's only used inside `aks` package so I thought we can make it private

Comment (alexander-demicev) [reply pkg/aks/check.go:116]: I've just done a quick search across rancher org https://github.com/search?q=org%3Arancher+CheckLogAnalyticsWorkspaceForMonitoring&type=code
it should be safe to make it private

Comment (richardcase) [reply pkg/aks/check.go:116]: Could we consider adding apidiff in the future to check for public api changes?

File: pkg/aks/check_test.go [hunk 2]
```diff
@@ -24,3 +31,83 @@ func Test_generateUniqueLogWorkspace(t *testing.T) {
...
+		workplacesClientMock.EXPECT().CreateOrUpdate(ctx, gomock.Any(), gomock.Any(), gomock.Any()).Return(operationalinsights.WorkspacesCreateOrUpdateFuture{}, errors.New("error"))
+
+		_, err := checkLogAnalyticsWorkspaceForMonitoring(ctx,
+			workplacesClientMock,
+			"eastus", "workspaceResourceGroup", "", "workspaceName")
+
+		Expect(err).To(HaveOccurred())
+	})
+
+	It("should fail if AsyncCreateUpdateResult returns error", func() {
```
Comment (richardcase) [inline pkg/aks/check_test.go:102]: Maybe in the future we could add a test where it fails initially but on the next poll it passes, a nice to have for the future.

File: pkg/aks/create_test.go [hunk 3]
```diff
@@ -14,45 +17,422 @@ import (
...
+		mockController       *gomock.Controller
+		workplacesClientMock *mock_services.MockWorkplacesClientInterface
+		clusterSpec          *aksv1.AKSClusterConfigSpec
+		cred                 *Credentials
+	)
+
+	BeforeEach(func() {
+		mockController = gomock.NewController(GinkgoT())
+		workplacesClientMock = mock_services.NewMockWorkplacesClientInterface(mockController)
+		clusterSpec = &aksv1.AKSClusterConfigSpec{
```
Comment (richardcase) [inline pkg/aks/create_test.go:73]: nit: might be nice to separate this out into a function.

**Review by richardcase — APPROVED** [review]
Looks good. Just 1 nit and a couple of comments.

File: pkg/aks/check.go (hunk 1 above)
Comment (alexander-demicev) [reply pkg/aks/check.go:116]: Yes, I can add apidiff check

File: pkg/aks/update.go [hunk 4]
```diff
@@ -5,8 +5,161 @@ import (
...
+		updateCluster(aksCluster, managedCluster),
+	)
+
+	return err
+}
+
+func updateCluster(aksCluster containerservice.ManagedCluster, managedCluster containerservice.ManagedCluster) containerservice.ManagedCluster {
+	/*
+		The following fields are managed in Rancher but are NOT configurable on update
+		- Name
```
Comment (richardcase) [inline pkg/aks/update.go:51]: If these are changed should we log a warning?

**Review by richardcase — APPROVED** [review]
LGTM.

One comment around fields that can't be changed on update.

**Pull request merged**